courses/views.py

```diff
@@ -139,29 +139,30 @@ def upstream_error(msg=None):
         }
         log.error(err['error'])
         return Response(err, status=502)
+    data = request.data
     missing = {
-        x for x in ('master_course_id', 'user_email', 'total_seats', 'display_name')
-        if x not in request.POST
+        key for key in ('master_course_id', 'user_email', 'total_seats', 'display_name')
+        if key not in data
     }
     if missing:
         raise serializers.ValidationError(detail={
             'error': 'You did not supply required POST argument(s): {}'.format(','.join(missing)),
         })
 
-    course = get_object_or_404(Course, uuid=request.POST['master_course_id'])
+    course = get_object_or_404(Course, uuid=data['master_course_id'])
     access_token = get_access_token(course.edx_instance)
-    user_email = request.POST['user_email']
+    user_email = data['user_email']
 
     payload = {
         'master_course_id': course.course_id,
         'coach_email': user_email,
-        'max_students_allowed': request.POST['total_seats'],
-        'display_name': request.POST['display_name'],
+        'max_students_allowed': data['total_seats'],
+        'display_name': data['display_name'],
     }
-    course_modules = request.POST.getlist('course_modules')
-    if course_modules:
-        course_modules_qs = Module.objects.filter(course=course, uuid__in=course_modules)
-        if course_modules_qs.count() != len(course_modules):
+    course_modules = data.get('course_modules')
+    if course_modules is not None:
```

Comment:
Just an explanation: the reason why I did not put `is not None` is because I did want to take in account the case when the client sends a `[]`content. In this case there is no reason to make the following query, even if everything still works.